docs/extended_api/synchronization_primitives/atomic_ref.md

```diff
@@ -0,0 +1,101 @@
+---
+grand_parent: Extended API
+parent: Synchronization Primitives
+nav_order: 1
+---
+
+# `cuda::atomic_ref`
+
+Defined in header `<cuda/atomic>`:
+
+```cuda
+template <typename T, cuda::thread_scope Scope = cuda::thread_scope_system>
+class cuda::atomic_ref;
+```
+
+The class template `cuda::atomic_ref` is an extended form of [`cuda::std::atomic_ref`]
+  that takes an additional [`cuda::thread_scope`] argument, defaulted to
+  `cuda::std::thread_scope_system`.
+It has the same interface and semantics as [`cuda::std::atomic_ref`], with the
+  following additional operations.
+This class additionally deviates from the standard by being backported to C++11.
+
+## Limitations
+
+`cuda::atomic_ref<T>` and `cuda::std::atomic_ref<T>` may only be instantiated with types sized between 4 and 8 bytes.
+
+Constructing multiple `cuda::std::atomic_ref<T, Sco>` with the same reference, but different values for `Sco` is illegal.
+
+`cuda::std::atomic_ref<T>` may be used with structures, but the behavior of performing atomic operations through it on objects with padding bytes is undefined.
```

Comment:
`cuda::std::atomic_ref<T>` or `cuda::atomic_ref<T>` or both?

Reply:
Good point. 

So I've suggested above to change the line 27. But that won't fully fix this issue. @wmaxey @griwes :
* Does `cuda::std::atomic_ref<T>` take a `Scope` template parameter? I thought it shouldn't do that.
* If not, then do we guarantee that `cuda::std::atomic_ref<T>` is an alias for `cuda::atomic_ref<T, Scope>`? 
   * If its an alias, then my wording above fixes this I think.
   * If we want to reserve the right to provide a completely different implementation for `cuda::std::atomic_ref<T>` vs `cuda::atomic_ref<T, cuda::thread_scope_system>`, then we probably also need to state here that applications shall not concurrently reference the same object or subobjects thereof with both `cuda::std::atomic_ref` and `cuda::atomic_ref`.